File: src/main/java/com/lessmarkup/dataobjects/AttachedFile.java
package com.lessmarkup.dataobjects;

import com.lessmarkup.interfaces.data.AbstractDataObject;
import com.lessmarkup.interfaces.recordmodel.RequiredField;

public class AttachedFile extends AbstractDataObject {
    private String uniqueId;
    private String fileName;
    private String contentType;
    private byte[] data;

    @RequiredField
    public String getUniqueId() {
        return uniqueId;
    }
    public void setUniqueId(String uniqueId) {
        this.uniqueId = uniqueId;
    }

    @RequiredField
    public String getFileName() {
        return fileName;
    }
    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    @RequiredField
    public String getContentType() {
        return contentType;
    }
    public void setContentType(String contentType) {
        this.contentType = contentType;
    }

    @RequiredField
    public byte[] getData() {
        return data;
    }
    public void setData(byte[] data) {
        this.data = data;
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/AttachedFile.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

import com.lessmarkup.interfaces.annotations.RequiredField
import com.lessmarkup.interfaces.data.{BinaryData, DataObject}

class AttachedFile extends DataObject {
  @RequiredField
  var uniqueId: String = null
  @RequiredField
  var fileName: String = null
  @RequiredField
  var contentType: String = null
  @RequiredField
  var data: BinaryData = null
}
